front_end/content.py

```diff
@@ -864,7 +864,8 @@ def get_submissions_basic(self, course_id, assignment_id, exercise_id, user_id):
                  ORDER BY submission_id DESC'''
 
         for submission in self.fetchall(sql, (int(course_id), int(assignment_id), int(exercise_id), user_id,)):
-            submissions.append([submission["submission_id"], submission["date"].strftime("%a, %d %b %Y %H:%M:%S UTC"), submission["passed"]])
+            partner_name = self.get_user_info(submission["partner_id"])["name"] if submission["partner_id"] else None
```

Comment:
Although you could use the get_user_info function, we want to reduce the number of times that we hit the database. By instead using an INNER JOIN in the above code, we can reduce the number of hits to the database drastically. The current approach would hit the database for every submission that the user had made for a given exercise.